### PR TITLE
Filter api output event endpoint

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

* Add tests for filter output event endpoint
* Update existing tests to use datasetID instead of a static value

### How to review

Test against https://github.com/ONSdigital/dp-filter-api/pull/98

### Who can review

Anyone

### CODE REVIEW

File: publishing/filterAPI/post_filter_output_event_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+			os.Exit(1)
+		}
+
+		Convey("When an authorised request with invalid json body is sent to update filter output", func() {
+			Convey("Then fail to update filter output and return status bad request (400)", func() {
+
+				filterAPI.PUT("/filter-outputs/{filter_output_id}", filterOutputID).
+					WithHeader(serviceAuthTokenName, serviceAuthToken).
+					WithBytes([]byte(`{`)).
+					Expect().Status(http.StatusBadRequest).Body().Contains("Bad request - Invalid request body\n")
```
Comment (nshumoogum) [inline publishing/filterAPI/post_filter_output_event_test.go]: No need to check for newline

File: publishing/filterAPI/post_filter_output_event_test.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+			})
+		})
+
+		Convey("When an authorised request is made to update filter output and json body contains dimensions or an instance id", func() {
+			Convey("Then fail to update filter output and return status forbidden (403)", func() {
+
+				filterAPI.PUT("/filter-outputs/{filter_output_id}", filterOutputID).
+					WithHeader(serviceAuthTokenName, serviceAuthToken).
+					WithBytes([]byte(GetValidPUTFilterOutputWithDimensionsJSON())).
+					Expect().Status(http.StatusForbidden).Body().Contains("Forbidden from updating the following fields: [dimensions]\n")
```
Comment (nshumoogum) [inline publishing/filterAPI/post_filter_output_event_test.go]: Again no need to check response contains new line

File: publishing/filterAPI/post_filter_output_event_test.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+				filterAPI.PUT("/filter-outputs/{filter_output_id}", filterOutputID).
+					WithHeader(serviceAuthTokenName, serviceAuthToken).
+					WithBytes([]byte(`{"instance_id": "1234"}`)).
+					Expect().Status(http.StatusForbidden).Body().Contains("Forbidden from updating the following fields: [instance_id]")
```
Comment (nshumoogum) [inline publishing/filterAPI/post_filter_output_event_test.go:151]: Again no need to check response contains new line

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: publishing/filterAPI/post_filter_output_event_test.go (hunk 1 above)
Comment (CarlHembrough) [reply publishing/filterAPI/post_filter_output_event_test.go]: 👍

File: publishing/filterAPI/post_filter_output_event_test.go (hunk 2 above)
Comment (CarlHembrough) [reply publishing/filterAPI/post_filter_output_event_test.go]: 👍

File: publishing/filterAPI/post_filter_output_event_test.go (hunk 3 above)
Comment (CarlHembrough) [reply publishing/filterAPI/post_filter_output_event_test.go:151]: No newline here 😄

Comment (nshumoogum) [reply publishing/filterAPI/post_filter_output_event_test.go:151]: sorry

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**